### ISSUE TITLE
R client installation fails: No module named 'pip'

### ISSUE THREAD

**Issue by attilagk** [Sage-Bionetworks/synapser#151]
I already installed the Synapse R client successfully on another machine running R version 3.4.1 (2017-06-30) on x86_64, linux-gnu system.  But installation fails on the current machine, which is also a x86_64, linux-gnu system but is running R version 3.4.4 (2018-03-15).

I run the code for installation specified [in the Getting Started doc](http://docs.synapse.org/articles/getting_started.html#installing-synapse-clients) and I get the following error:
```
No module named 'pip'
Error in pyExec("import pip") : 
  An error has occurred while executing Python code. See traceback above.
```
 But the latest pip *is* installed on the system.  And I verified that `import pip` worked in the python REPL shell.  And on the present machine the Python client could be installed without any problems.  Any ideas what could cause this issue?

**Comment by kimyen** [comment]
@attilagk, could you provide the output of `sessionInfo()`?

**Comment by kimyen** [comment]
If the error you see includes the following:
```
No module named 'pip'
Error in pyExec("import pip") :
An error has occurred while executing Python code. See traceback above.
Execution halted
ERROR: Missing system dependencies. Please make sure that your machine has the required dependencies listed in the SystemRequirements field of the DESCRIPTION file.
ERROR: configuration failed for package ‘synapser’
```
Please check the DESCRIPTION file and make sure that your machine has sufficient dependencies.

**Comment by kimyen** [comment]
@attilagk 
Also, while helping a colleague on a similar problem, I created an AMI which has synapser installed with R 3.4.4 running under Ubuntu 16.04.4 LTS: ami-6d26fa12 (N.Virginia). Please feel free to start with this, and let us know if this unblocks you.

**Comment by attilagk** [comment]
@kimyen , thank you, below is the session info
```
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.4 LTS

Matrix products: default
BLAS: /usr/lib/openblas-base/libblas.so.3
LAPACK: /usr/lib/libopenblasp-r0.2.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.20     setwidth_1.0-4 colorout_1.0-2

loaded via a namespace (and not attached):
[1] compiler_3.4.4 tools_3.4.4   
```

**Comment by kimyen** [comment]
One thing I realized while creating the AMI is that a synapser's dependencies, PythonEmbedInR, needs to be installed at the time the system dependencies is installed. 
This is how I check the system dependencies on Ubuntu:
```
ldconfig -v -p -N | grep libssl
	libssl3.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libssl3.so
	libssl.so.1.0.0 (libc6,x86-64) => /lib/x86_64-linux-gnu/libssl.so.1.0.0
	libssl.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libssl.so

ldconfig -v -p -N | grep libcurl
	libcurl.so.4 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcurl.so.4
	libcurl-gnutls.so.4 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.4

dpkg --get-selections | grep zlib
zlib1g:amd64					install
zlib1g-dev:amd64				install
```
Then I make sure that PythonEmbedInR is uninstalled (and gone from /R/site-packages). 
Finally, I install synapser again. Installing synapser should automatically installing its dependencies including PythonEmbedInR. Now that all dependencies are in place, the pip problem should go away.

I created a Jira issues to track a fix on moving this problem to PythonEmbedInR and solve it there. So that users would not be confused about this when installing synapser.

Please let me know if you are unblocked.

**Comment by attilagk** [comment]
I removed the PythonEmbedInR package and tried to install synapser again but installation failed with the following error.
```
Traceback (most recent call last):

  File "<string>", line 1, in <module>

  File "/tmp/Rtmp6oYX1f/R.INSTALL7d767030e19d/synapser/inst/python/installPythonClient.py", line 81, in main

    
import synapseclient
  File "/tmp/Rtmp6oYX1f/R.INSTALL7d767030e19d/synapser/inst/lib/python3.5/site-packages/synapseclient-1.7.5-py3.5.egg/synapseclient/__init__
.py", line 308, in <module>

    
from .client import Synapse, login
  File "/tmp/Rtmp6oYX1f/R.INSTALL7d767030e19d/synapser/inst/lib/python3.5/site-packages/synapseclient-1.7.5-py3.5.egg/synapseclient/client.p
y", line 76, in <module>

    
from . import cache
  File "/tmp/Rtmp6oYX1f/R.INSTALL7d767030e19d/synapser/inst/lib/python3.5/site-packages/synapseclient-1.7.5-py3.5.egg/synapseclient/cache.py
", line 28, in <module>

    
import synapseclient.utils as utils
  File "/tmp/Rtmp6oYX1f/R.INSTALL7d767030e19d/synapser/inst/lib/python3.5/site-packages/synapseclient-1.7.5-py3.5.egg/synapseclient/utils.py
", line 13, in <module>

    
from future.utils import implements_iterator
ImportError
: 
No module named 'future'
Error in pyExec(command) : 
  An error has occurred while executing Python code. See traceback above.
Execution halted
ERROR: configuration failed for package ‘synapser’
* removing ‘/home/attila/R/library/synapser’

The downloaded source packages are in
        ‘/tmp/Rtmp4wmicW/downloaded_packages’
Warning message:
In install.packages("synapser", repos = c("https://sage-bionetworks.github.io/ran",  :
  installation of package ‘synapser’ had non-zero exit status
```

**Comment by kimyen** [comment]
@attilagk I think the main error message is:
```
[Errno -3] Temporary failure in name resolution
```
To understand this problem, we wrapped our Python client and generate R code. During installation, we installed a separate Python environment in the package PythonEmbedInR. The Python client and its dependencies will be installed there. The purpose of having an isolated Python environment is so that synapser would not interfere or be affected by the existing Python environment in a user's machine.

It looks like you are in a state where `backports.csv` name already exist in the isolated Python environment. I would suggest to solve the problem by removing PythonEmbedInR completely and rerun your installation command (with appropriate permissions.) 

I will take note about this and discuss with my colleagues to see what we can do to improve user's experience here.

**Comment by kimyen** [comment]
We released synapser 0.2 to [Sage's RAN](https://sage-bionetworks.github.io/ran/). With the new version, when users does not have sufficient system dependencies, PythonEmbedInR will fail to install. This prevent users to get into the state where they have to manually remove PythonEmbedInR to proceed.
